### ISSUE TITLE
Enable internal feeds during internal builds, but leave them disabled otherwise

### ISSUE THREAD

**Issue by dagood** [dotnet/source-build#1696]
Currently repos have internal feeds enabled by default. For internal tarball production builds, we set up authentication to use them. Otherwise, we have build tasks that delete them.

In the future (https://github.com/dotnet/arcade/issues/5890), repos will have internal feeds present but disabled by default. We'll need to have internal tarball production builds enable them and still authenticate to them. Otherwise, we will need to leave them alone in the disabled state.

We can probably do this infra work ahead of time and have the build tasks determine what needs to happen based on lines like this being present:

```xml
<disabledPackageSources>
    <add key="darc-int-dotnet-core-setup-0c2e69ca" value="true" />
</disabledPackageSources>
```

**Comment by mmitche** [comment]
@dagood Does source-build not use SetupNuGetSources.sh right now to add creds?

**Comment by dagood** [comment]
> Does source-build not use SetupNuGetSources.sh right now to add creds?

Correct. We use the NuGet auth provider with ordinary auth environment, like Core-Setup (last I saw it). We weren't set up for internal builds properly for the 3.1 servicing chain initially, so I implemented it this way in 3.1.106 because it was relatively simple and got rid of some not ideal workarounds we were using.
